app/controllers/katello/api/v2/products_controller.rb

```diff
@@ -41,8 +41,7 @@ def index_relation
       query = query.where(:name => params[:name]) if params[:name]
       query = query.enabled if params[:enabled]
       query = query.where(:id => @activation_key.products) if @activation_key
-      query = query.where(:id => @system.products) if @system
-      query = query.where(:id => Pool.find_by_id!(params[:subscription_id]).products) if params[:subscription_id]
+      query = query.where(:id => Pool.find_by_cp_id(params[:subscription_id]).products) if params[:subscription_id]
```

Comment:
something similar to a uuid like ff8080814f708384014f70acb98d0f5c which is stored as cp_id in pools and subscriptions, the API is currently taking this as the id. I will look into supporting both.